### PR TITLE
cmake-utils.eclass: Split multi-arg CC/CXX/FC correctly, #542530 [take 3]

### PR DESCRIPTION
**Pull request by mgorny**
Split multi-argument CC/CXX/FC into a CMake array consisting of one
element containing the compiler path (name) and the other containing all
command-line options, space-separated. This is how CMake splits
the environment variables CC/CXX/FC internally.

The alternative would be to set CC/CXX/FC directly, and let CMake handle
the splitting. However, changing that had unforeseen consequences like
assembler not being set correctly (#601292) which in turn was impossible
to set via environment variables due to bugs in CMake.

Therefore, splitting the values into CMAKE_*_COMPILER correctly seems
the simplest and safest way forward, at least until we can rely on fixed
CMake versions being sufficiently deployed.

@gentoo/kde

### CODE REVIEW

**Comment by mgorny** [comment]
@toralf, could you give this patch a tinderbox run? Packages using cmake-utils are of interest, preferably with ABI_X86="32 64" for better coverage.

File: eclass/cmake-utils.eclass [hunk 1]
```diff
@@ -523,11 +523,18 @@ enable_cmake-utils_src_configure() {
 		SET (CMAKE_Fortran_COMPILE_OBJECT "<CMAKE_Fortran_COMPILER> <DEFINES> ${includes} ${FCFLAGS} <FLAGS> -o <OBJECT> -c <SOURCE>" CACHE STRING "Fortran compile command" FORCE)
 	_EOF_
 
+	local myCC=$(tc-getCC) myCXX=$(tc-getCXX) myFC=$(tc-getFC)
+
+	# !!! IMPORTANT NOTE !!!
+	# Single slash below is intentional. CMake is weird and wants the
+	# CMAKE_*_VARIABLES split into two elements: the first one with
+	# compiler path, and the second all with all command-line options,
```
Comment (Arfrever) [inline eclass/cmake-utils.eclass]: Probably s/second all/second one/

Comment (toralf) [inline eclass/cmake-utils.eclass]: applied
$> patch -p1 < /root/1dbb40df4d570c8fed6ff564b312b7f1ec07cae6.patch
here onto /usr/portage of the tinderbox

Comment (mgorny) [inline eclass/cmake-utils.eclass]: @Arfrever, thanks, fixed on my local branch.

Comment (toralf) [inline eclass/cmake-utils.eclass]: builds fine at all tindrebox images

furthermore I checked at: amd64-desktop-unstable_20161202-095146/var/log/emerge.log

Mon Dec  5 11:28:35 2016 >>> dev-games/simgear-2016.3.1
simgear: Mon Dec  5 11:17:47 2016: 10 minutes, 48 seconds
simgear: 1 times

having ABI_X86="32 64"

**Comment by mgorny** [comment]
Merged.

**Pull request closed without merging**